### PR TITLE
Ankit | Test | Version 5 - Support of Payment history and overdue in VAT report 

### PR DESCRIPTION
**Pull request by ankitdubeyoffice**
https://app.clickup.com/t/86cweab7c
https://app.clickup.com/t/86cweaquz
https://app.clickup.com/t/86cweaz04
https://app.clickup.com/t/86cweb0tf
https://app.clickup.com/t/86cweb63w
https://app.clickup.com/t/86cweb8y0

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts [hunk 1]
```diff
@@ -66,25 +64,23 @@ export class VatLiabilitiesPayments implements OnInit, OnDestroy {
...
     /** Holds current table columns */
     public displayColumns: string[] = [];
     /** Observable to store true if API Call is in progress */
     public liabilityPaymentListInProgress$ = this.componentStore.select(state => state.liabilityPaymentListInProgress);
     /** Holds true if user in vat-payment */
     public isPaymentMode: boolean;
-    /** Company base currency symbol */
-    public baseCurrencySymbol: string = "";
+    /** Stores the current company */
+    public activeCompany: any={};
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]:
```suggestion
    public activeCompany: any = {};
```

File: apps/web-giddh/src/app/vat-report/utility/vat.report.store.ts [hunk 2]
```diff
@@ -23,11 +25,14 @@ export class VatReportComponentStore extends ComponentStore<VatReportState> {
     constructor(
         private toaster: ToasterService,
         private vatService: VatService,
-        private gstReconcileService: GstReconcileService
+        private gstReconcileService: GstReconcileService,
+        private store: Store<AppState>,
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/vat-report/utility/vat.report.store.ts]:
```suggestion
        private store: Store<AppState>
```

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts [hunk 3]
```diff
@@ -78,14 +81,20 @@ export class TaxSidebarComponent implements OnInit, OnDestroy {
     public isUSCompany: boolean;
     /** Holds active company information */
     public activeCompany: any;
+    /** True if Current branch has Tax Number */
+    public hasTaxNumber: boolean = false;
+    /** Stores the current branch */
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts]:
```suggestion
    /** Holds current branch  information */
```

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts [hunk 4]
```diff
@@ -78,14 +81,20 @@ export class TaxSidebarComponent implements OnInit, OnDestroy {
     public isUSCompany: boolean;
     /** Holds active company information */
     public activeCompany: any;
+    /** True if Current branch has Tax Number */
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts]:
```suggestion
    /** True if current company or branch has tax number */
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts [hunk 5]
```diff
@@ -66,25 +64,23 @@ export class VatLiabilitiesPayments implements OnInit, OnDestroy {
...
-    /** Company base currency symbol */
-    public baseCurrencySymbol: string = "";
+    /** Stores the current company */
+    public activeCompany: any = {};
     /** This will hold the value out/in to open/close setting sidebar popup */
     public asideGstSidebarMenuState: string = 'in';
 
     constructor(
         private activatedRoute: ActivatedRoute,
-        private gstReconcileService: GstReconcileService,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts:83]: Remove also in import for VatService, GstReconcileService

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts (hunk 5 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts:83]: Done

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts (hunk 4 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts]: Done

File: apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/shared/tax-sidebar/tax-sidebar.component.ts]: Done

File: apps/web-giddh/src/app/vat-report/utility/vat.report.store.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/utility/vat.report.store.ts]: Done

File: apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vat-report/vat-liabilities-payments/vat-liabilities-payments.component.ts]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**